File: clients/algoliasearch-client-php/lib/Model/Recommend/RemoveWordsIfNoResults.php
<?php

namespace Algolia\AlgoliaSearch\Model\Recommend;

/**
 * RemoveWordsIfNoResults Class Doc Comment
 *
 * @category Class
 * @description Selects a strategy to remove words from the query when it doesn&#39;t match any hits.
 *
 * @package Algolia\AlgoliaSearch
 */
class RemoveWordsIfNoResults
{
    /**
     * Possible values of this enum
     */
    const NONE = 'none';

    const LAST_WORDS = 'lastWords';

    const FIRST_WORDS = 'firstWords';

    const ALL_OPTIONAL = 'allOptional';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public static function getAllowableEnumValues()
    {
        return [
            self::NONE,
            self::LAST_WORDS,
            self::FIRST_WORDS,
            self::ALL_OPTIONAL,
        ];
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/com/algolia/model/recommend/RemoveWordsIfNoResults.java
package com.algolia.model.recommend;

import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

/** Selects a strategy to remove words from the query when it doesn't match any hits. */
@JsonAdapter(RemoveWordsIfNoResults.Adapter.class)
public enum RemoveWordsIfNoResults {
  NONE("none"),

  LAST_WORDS("lastWords"),

  FIRST_WORDS("firstWords"),

  ALL_OPTIONAL("allOptional");

  private final String value;

  RemoveWordsIfNoResults(String value) {
    this.value = value;
  }

  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  public static RemoveWordsIfNoResults fromValue(String value) {
    for (RemoveWordsIfNoResults b : RemoveWordsIfNoResults.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }

  public static class Adapter extends TypeAdapter<RemoveWordsIfNoResults> {

    @Override
    public void write(
      final JsonWriter jsonWriter,
      final RemoveWordsIfNoResults enumeration
    ) throws IOException {
      jsonWriter.value(enumeration.getValue());
    }

    @Override
    public RemoveWordsIfNoResults read(final JsonReader jsonReader)
      throws IOException {
      String value = jsonReader.nextString();
      return RemoveWordsIfNoResults.fromValue(value);
    }
  }
}
